presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlockBuilder.java

```diff
@@ -172,12 +197,27 @@ public BlockBuilder appendNull()
 
     private void entryAdded(boolean isNull)
     {
+        if (!initialized) {
+            initializeCapacity();
+        }
         valueIsNull.appendByte(isNull ? 1 : 0);
 
         positionCount++;
         blockBuilderStatus.addBytes(Byte.BYTES + fixedSize);
     }
 
+    private void initializeCapacity()
+    {
+        if (positionCount != 0 || currentEntrySize != 0) {
+            throw new IllegalStateException(getClass().getSimpleName() + " was used before initialization");
+        }
+        initialized = true;
```

Comment:
It would be more natural to set `initalized = true` as last statement in initalize() method